subprojects/core/src/main/java/org/gradle/api/internal/tasks/NonCacheableTaskOutputPropertySpec.java

```diff
@@ -26,23 +26,29 @@
 public class NonCacheableTaskOutputPropertySpec extends TaskOutputsDeprecationSupport implements TaskOutputFilePropertySpec {
 
     private final CompositeTaskOutputPropertySpec parent;
+    private final int index;
     private final FileCollection files;
 
-    public NonCacheableTaskOutputPropertySpec(String taskName, CompositeTaskOutputPropertySpec parent, FileResolver resolver, Object paths) {
+    public NonCacheableTaskOutputPropertySpec(String taskName, CompositeTaskOutputPropertySpec parent, int index, FileResolver resolver, Object paths) {
         this.parent = parent;
+        this.index = index;
         this.files = new TaskPropertyFileCollection(taskName, "output", this, resolver, paths);
     }
 
     @Override
     public String getPropertyName() {
-        return parent.getPropertyName();
+        return parent.getPropertyName() + "$" + index;
     }
 
     @Override
     public OutputType getOutputType() {
         return parent.getOutputType();
     }
 
+    public CompositeTaskOutputPropertySpec getParent() {
+        return parent;
+    }
```

Comment:
This is only used to refer to the parent property name now. I think it would be better to make it explicit by having a call instead like `getOriginalPropertyName()` or something.

Reply:
I added `getOriginalPropertyName`.